includes/wf_crm_webform_base.inc

```diff
@@ -527,10 +528,22 @@ abstract class wf_crm_webform_base {
       );
     }
     $file = wf_crm_apivalues('file', 'get', $val);
+    $entity_id = '';
+    if ($entity && strpos($fieldName, 'custom_') === 0 ) {
+      switch ( $entity ) {
+      case 'contact':
+        $entity_id = $_GET['cid1'];
+        break;
+      case 'activity':
+        $entity_id = $_GET['activity1'];
+        break;
+      }
```

Comment:
$_GET is not a reliable source for ids and I don't think you can count on the number always being 1. Try replacing this switch statement with something like

    $entity_id = $this->ent[$entity][$n]['id']

Although you'll need to somehow get the value for `$n`